core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheFileSystem.java

```diff
@@ -23,14 +24,16 @@
 public class LocalCacheFileSystem extends DelegatingFileSystem {
 
   private final LocalCacheManager mLocalCacheManager;
+  private final FileSystemContext mFsContext;
 
   /**
    * @param fs a FileSystem instance to query on local cache miss
    */
-  public LocalCacheFileSystem(FileSystem fs) {
+  public LocalCacheFileSystem(FileSystem fs, FileSystemContext fsContext) {
```

Comment:
I don't feel very good about this constructor. It seems counter-intuitive to have the fsContext as an argument here.

There is already an fsContext embededed within the `fs` object. Is there a better way to access it?

Reply:
I don't see a way to access it unless we expose it from the `FileSystem` interface.